### PR TITLE
Fix most of the scrolling bugs when navigating

### PR DESCRIPTION
**Pull request by DanielHeath**

### CODE REVIEW

**Comment by DanielHeath** [comment]
It looks like this fixes scroll for `forward` and `back` but not `<Link>`

File: src/pages/ResultsListPage.js [hunk 1]
```diff
@@ -101,8 +101,11 @@ class ResultsListPage extends BaseResultsPage {
     }
 
     renderPersonalisationLink(): ReactElement {
+        const current = this.props.location.pathname;
+        const separator = current.endsWith("/") ? "" : "/";
```
Comment (danni) [inline src/pages/ResultsListPage.js:105]: There's a method for this in Node URL.

Comment (DanielHeath) [inline src/pages/ResultsListPage.js]: And also in the browser?

Comment (DanielHeath) [inline src/pages/ResultsListPage.js]: Ah right - Location is a URL object so that should be fine.

Comment (DanielHeath) [inline src/pages/ResultsListPage.js:105]: ergh, relative URLs lose the last segment of the path when using `url.resolve`. Don't think that will work.

Comment (danni) [inline src/pages/ResultsListPage.js:105]: Do we want the last segment then?

Comment (DanielHeath) [inline src/pages/ResultsListPage.js:105]: Yes - we need eg `/category/housing/personalise/summary`, not `/category/personalise/summary`

Comment (danni) [inline src/pages/ResultsListPage.js:105]: Hmm. Could we parse the URL, append to the path array and reoutput?

Comment (DanielHeath) [inline src/pages/ResultsListPage.js:105]: The path isn't an array - we could split it on '/' and remove all the empty strings from the resulting array, but that also sounds like hacks.

**Comment by danni** [comment]
++

**Pull request merged**